src/openforms/variables/models.py

```diff
@@ -126,12 +127,13 @@ def request_arguments(self, context: DataMapping) -> dict:
         #
         # extra knowledge not in the RFC: latin1 is a different name for ISO-8859-1
 
+        ctx = recursive_apply(context, str, transform_leaf=True)
```

Comment:
```suggestion
        # Explicitly cast values to strings to avoid localization
        ctx = recursive_apply(context, str, transform_leaf=True)
```